### PR TITLE
iterator for all the tensor elements other than parameters of nn.module

### PR DESCRIPTION
**Pull request by ratmcu**
new internal iterator could be extended with tensor element buffers (parameters, buffers ) in the nn.Module. this update holds all the internal buffers inside the nn.Module.

### CODE REVIEW

**Comment by ratmcu** [comment]
@iamtrask do you think at the moment this modification would be suffice for the movement of non parametric tensor elements as demanded by https://github.com/OpenMined/PySyft/issues/3236 ? . I was hoping until we elegantly add a hook for the buffers underlying the nn.Module and Optimizers(which is not using a buffer definition at the moment from pytorch, which I hope we can add a feature in pytorch.)  we can keep updating the list of iterators from the nn.Module  if a new iterator of elements comes up.

**Comment by iamtrask** [comment]
Still failing codecov - but if you can get that to pass (above 95%) we're good :)

File: syft/frameworks/torch/hook/hook.py [hunk 1]
```diff
@@ -601,8 +617,12 @@ def module_send_(nn_self, *dest, force_send=False, **kwargs):
             if module_is_missing_grad(nn_self):
                 create_grad_objects(nn_self)
 
-            for p in nn_self.parameters():
-                p.send_(*dest, **kwargs)
+            # for p in nn_self.parameters():
+            #     p.send_(*dest, **kwargs)
```
Comment (karlhigley) [inline syft/frameworks/torch/hook/hook.py:621]: Looks like the code below replaces these lines, so probably good to remove them.

**Comment by karlhigley** [comment]
(@imskr This is another example where the main code coverage tests pass but Codecov still fails.)

**Comment by karlhigley** [comment]
Test failure is from an unrelated known flaky test.

**Pull request merged**